scala2.13/dist/pom.xml

```diff
@@ -49,28 +49,6 @@
         <rapids.source.jar.phase>none</rapids.source.jar.phase>
     </properties>
     <profiles>
-        <profile>
-            <id>noSnapshotsWithDatabricks</id>
-            <properties>
-                <included_buildvers>
-                    ${noSnapshot.buildvers},
-                    ${databricks.buildvers}
-                </included_buildvers>
-            </properties>
-        </profile>
-        <profile>
-            <id>noSnapshots</id>
```

Comment:
There are still references to these kind of profiles

```bash
~/gits/NVIDIA/spark-rapids.reviews$ git grep 'Psnapshots\|PnoSnapshots'
.github/workflows/mvn-verify-check/populate-daily-cache.sh:            -DforceStdout -PnoSnapshots -q | tr -d ',' | \
CONTRIBUTING.md:1. Build for all Apache Spark versions and CDH with no SNAPSHOT versions of Spark, only released. Use `-PnoSnapshots`.
CONTRIBUTING.md:2. Build for all Apache Spark versions and CDH including SNAPSHOT versions of Spark we have supported for. Use `-Psnapshots`.
CONTRIBUTING.md:4. Build for all Apache Spark versions, CDH and Databricks including SNAPSHOT versions of Spark we have supported for. Use `-PsnapshotsWithDatabricks`
CONTRIBUTING.md:mvn -pl dist -PnoSnapshots package -DskipTests
CONTRIBUTING.md:mvn package -pl dist -PnoSnapshots -Ddist.jar.compress=false -Drapids.jni.unpack.skip
jenkins/version-def.sh:# Psnapshots: snapshots + noSnapshots
jenkins/version-def.sh:# PnoSnapshots: noSnapshots only 
```

Should we leave the profiles in place with them merely defining the right `expression` value.

Also I am not clear how to build without buildall. Before this PR It is easy to install each shim then run dist with the profile like -PnoSnapshots

What should I do now?

```
mvn -pl dist package -am verify -Dexpression=no_snapshots.buildvers
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.1.0:run (shimplify-shim-sources) on project rapids-4-spark-parent_2.12: An Ant BuildException has occured: Traceback (most recent call last):
[ERROR]   File "<script>", line 195, in <module>
[ERROR] AttributeError: 'NoneType' object has no attribute 'split'
[ERROR] 
[ERROR] around Ant part ...<shimplify if="shimplify" />... @ 8:33 in /home/user/gits/NVIDIA/spark-rapids.reviews/target/spark320/antrun/build-main.xml
[ERROR] -> [Help 1]
```

Reply:
I will have to clean up the references in the `CONTRIBUTING.md` once we decide on the final solution. I don't think we need profiles defining moving lists in the pom.xml because there are ways to build the project without using profiles. 

To build without `buildall` for versions 330 and 350, use the following commands
```
mvn initialize -pl sql-plugin-api -am
mvn -U install -DskipTests -Dbuildver=330 -Drat.skip=false -Dmaven.scaladoc.skip -Dmaven.scalastyle.skip=false -pl tools -am
mvn -U install -DskipTests -Dbuildver=350 -Drat.skip=false -Dmaven.scaladoc.skip -Dmaven.scalastyle.skip=false -pl tools -am
mvn package -rf aggregator -Dincluded_buildvers=330,350 -Dbuildver=330 -DskipTests -Dmaven.scaladoc.skip
```

Reply:
The profiles are handy for quickly testing the dist assembly logic using the shim aggregator artifacts pushed by the CI without having to rebuild them. Testing a few works fine with listing them explicitly. But If we need to test noSnapshotsWithDatabricks it is tedious and error-prone. On top of it, we cannot build the DBR shims locally